src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -457,14 +794,19 @@ class SegmentCleaner : public SegmentProvider {
   using init_segments_ret = init_segments_ertr::future<init_segments_ret_bare>;
   init_segments_ret init_segments();
 
-  get_segment_ret get_segment() final;
+  get_segment_ret get_segment(device_id_t id = 0) final;
```

Comment:
Does defaulting to an id of 0 make sense?  Are we guaranteed that there is a device id of 0?

Reply:
yes, I think so, I always set "device/block" to device id 0

Reply:
I don't think this makes sense.  The only user I see that doesn't pass an argument is journal.cc:43 in roll_journal_segment.  That usage only makes sense if you guarantee that the journal device always has ID 0.  I see no comment or code that enforces that requirement.

More generally, I don't think we want a caller that forgets to pass an argument to get the 0 device -- we want it to be a compile error.  If we need a call which selects a segment from a device based on a heuristic, we can add that, but it should be explicit.

Remove this default argument and modify that caller in journal.cc to pass the id of the SegmentManager that Journal has a reference to.

Reply:
ok, will do